### ISSUE TITLE
(GLFW3) glfwCreateWindow return a value even if canvas creation failed

### ISSUE THREAD

**Issue by gouletr** [kripken/emscripten#3764]
Calling `glfwCreateWindow` in GLFW3 Emsripten returns a pointer even if the canvas creation failed. This can be reproduced by trying to create a WebGL2 canvas while the browser doesn't support it. Ideally, it should return `nullptr` so that we can handle the error.

**Comment by gouletr** [comment]
The implementation in Emscripten looks like this:

```javascript
    createWindow: function(width, height, title, monitor, share) {
      var i, id;
      for (i = 0; i < GLFW.windows.length && GLFW.windows[i] !== null; i++);
      if (i > 0) throw "glfwCreateWindow only supports one window at time currently";

      // id for window
      id = i + 1;

      // not valid
      if (width <= 0 || height <= 0) return 0;

      if (monitor) {
        GLFW.requestFullScreen();
      } else {
        Browser.setCanvasSize(width, height);
      }

      // Create context when there are no existing alive windows
      for (i = 0; i < GLFW.windows.length && GLFW.windows[i] == null; i++);
      if (i == GLFW.windows.length) {
        var contextAttributes = {
          antialias: (GLFW.hints[0x0002100D] > 1), // GLFW_SAMPLES
          depth: (GLFW.hints[0x00021005] > 0),     // GLFW_DEPTH_BITS
          stencil: (GLFW.hints[0x00021006] > 0)    // GLFW_STENCIL_BITS
        }
        Module.ctx = Browser.createContext(Module['canvas'], true, true, contextAttributes);
      }

      // Get non alive id
      var win = new GLFW.Window(id, width, height, title, monitor, share);

      // Set window to array
      if (id - 1 == GLFW.windows.length) {
        GLFW.windows.push(win);
      } else {
        GLFW.windows[id - 1] = win;
      }

      GLFW.active = win;
      return win.id;
    },
```
Perhaps we should check for the value of Module.ctx before doing the rest of the work?

**Comment by kripken** [comment]
Yes, that sounds correct. Want to make a pull request?

**Comment by gouletr** [comment]
I've never written any javascript before, so I'm unsure about the syntax. But I can try! What's the return value type? Would it be `0` or `null` or something equivalent in javascript?

**Comment by kripken** [comment]
Looks like that return value goes back to the caller of `glfwCreateWindow`, so it should be the value that method should return, which I think in this case is a NULL pointer in C. That's just a 0 in JS for us.

**Comment by gouletr** [comment]
PR open : https://github.com/kripken/emscripten/pull/3768

**Comment by gouletr** [comment]
Fixed in commit e7aa8a2f0432c112209099c8c2fa63b0932acf0a.
